resolwe_bio/processes/alignment/star.py

```diff
@@ -505,17 +517,29 @@ def run(self, inputs, outputs):
                 ]
             )
 
+        gene_segments = Path(inputs.genome.output.index.path) / "geneInfo.tab"
         if inputs.t_coordinates.quant_mode:
-            gene_segments = Path(inputs.genome.output.index.path) / "geneInfo.tab"
             if not gene_segments.is_file() and not inputs.annotation:
                 self.error(
                     "Output in transcript coordinates requires genome annotation file."
                 )
 
-            star_params.extend(["--quantMode", "TranscriptomeSAM"])
+            if inputs.gene_counts:
+                star_params.extend(["--quantMode", "TranscriptomeSAM", "GeneCounts"])
+            else:
+                star_params.extend(["--quantMode", "TranscriptomeSAM"])
+
             if inputs.t_coordinates.single_end:
                 star_params.extend(["--quantTranscriptomeBan", "Singleend"])
 
+        elif inputs.gene_counts and not inputs.t_coordinates.quant_mode:
```

Comment:
`and not inputs.t_coordinates.quant_mode` is redundant as it is already checked with the first if statement.